pinot-core/src/main/java/com/linkedin/pinot/core/startree/StarTreeDataTable.java

```diff
@@ -211,12 +209,12 @@ public void remove() {
         public Pair<byte[], byte[]> next() {
           byte[] dimBuff = new byte[dimensionSizeInBytes];
           byte[] metBuff = new byte[metricSizeInBytes];
-          mappedByteBuffer.copyTo(pointer * totalSizeInBytes, dimBuff, 0, dimensionSizeInBytes);
-//          mappedByteBuffer.position(pointer * totalSizeInBytes);
-//          mappedByteBuffer.get(dimBuff);
+          copyTo(mappedByteBuffer, pointer * (long) totalSizeInBytes, dimBuff, 0, dimensionSizeInBytes);
```

Comment:
Are you sure "toDirectByteBuffer" returns an array? (not "toDirectByteBuffers", these two are different, and the return type array is used for longer than MAX_INT copy length, which is not necessary for our purpose).